consul/util.go

```diff
@@ -95,6 +95,30 @@ func ensurePath(path string, dir bool) error {
 	return os.MkdirAll(path, 0755)
 }
 
+// CanServersUnderstandProtocol checks to see if all the servers in the given
+// list understand the given protocol version or higher. If there are no servers
+// in the list then this will return false.
+func CanServersUnderstandProtocol(members []serf.Member, version uint8) (bool, error) {
+	numServers, numWhoGrok := 0, 0
+	for _, m := range members {
+		if m.Tags["role"] != "consul" {
+			continue
+		}
+		numServers++
+
+		vsn_str := m.Tags["vsn_max"]
+		vsn, err := strconv.Atoi(vsn_str)
+		if err != nil {
+			return false, err
+		}
+
+		if vsn >= int(version) {
```

Comment:
Should there also be a `<=` here for the `vsn_min`? Likely wouldn't see that any time soon I guess.